app/api/actions/github/route.ts

```diff
@@ -324,174 +323,210 @@ export async function POST(request: Request) {
           {
             owner,
             repo,
-            comment_id: botComment.id,
-            body: textToWrite,
+            issue_number: number,
+            headers: {
+              "X-GitHub-Api-Version": "2022-11-28",
+            },
           }
         );
-      } else {
-        // Post a new comment if no existing comment was found
-        await octokit
-          .request(
-            "POST /repos/{owner}/{repo}/issues/{issue_number}/comments",
+        console.info("comments.data.length", comments.data.length);
+        // Find our bot's comment
+        let botComment = comments.data.find((comment) => {
+          return comment.user.login.includes("watermelon-context");
+        });
+        if (botComment?.id) {
+          // Update the existing comment
+          await octokit.request(
+            "PATCH /repos/{owner}/{repo}/issues/comments/{comment_id}",
             {
               owner,
-              issue_number: number,
               repo,
+              comment_id: botComment.id,
               body: textToWrite,
             }
-          )
-          .then((response) => {
-            successPosthogTracking({
-              url: request.url,
-              email: user_email,
-              data: {
-                repo,
+          );
+        } else {
+          // Post a new comment if no existing comment was found
+          await octokit
+            .request(
+              "POST /repos/{owner}/{repo}/issues/{issue_number}/comments",
+              {
                 owner,
-                number,
-                action: req.action,
-                textToWrite,
-              },
-            });
-          })
-          .catch((error) => {
-            failedPosthogTracking({
-              url: request.url,
-              error: error.message,
-              email: req.email,
+                issue_number: number,
+                repo,
+                body: textToWrite,
+              }
+            )
+            .then((response) => {
+              successPosthogTracking({
+                url: request.url,
+                email: user_email,
+                data: {
+                  repo,
+                  owner,
+                  number,
+                  action: req.action,
+                  textToWrite,
+                },
+              });
+            })
+            .catch((error) => {
+              failedPosthogTracking({
+                url: request.url,
+                error: error.message,
+                email: req.email,
+              });
+              return console.error("posting comment error", error);
             });
-            return console.error("posting comment error", error);
-          });
+        }
+      } else {
+        textToWrite = `Your team has surpassed the free monthly usage. [Please click here](https://calendly.com/evargas-14/watermelon-business) to upgrade.`
```

Comment:
Why wrap the whole thing instead of only replacing the text?

Reply:
Same thing but if you're adamant about it I'll do it that way

Reply:
Pls review again

Reply:
Thank you, much easier to digest a 36 line change than a 300 one.

Reply:
100% thanks for your feedback